lib/message.rb

```diff
@@ -196,6 +198,18 @@ def sequence_segments(base=nil)
     end
   end
 
+  def enforce_encoding!
+    return unless self[:MSH] && POSSIBLE_ENCODINGS.include?(self[:MSH].charset)
```

Comment:
When this condition actually occurs: `POSSIBLE_ENCODINGS.include?(self[:MSH].charset)` it would be nice to have an error message like: `HL7 message is in #{self[:MSH].charset} which is not part of the allowed encodings for the standard (allowed encodings: #{POSSIBLE_ENCODINGS})`

Reply:
Throwing an error here might defeat the purpose. Perhaps allowing it pass without an exception until it gets to a point where it's impossible to proceed might be better. However, if we have an exhaustive list of encodings, then we might use this as it's just not sensible for the encoding to not be a subset of all possible encodings. What do you think?